controls/V-71971.rb

```diff
@@ -96,57 +87,53 @@
 # semanage login -m -s user_u <username>"
   tag "fix_id": "F-78323r1_fix"
 
-  # TODO should this be an onlyif or a force skip if not installed?
-  # Make sure semanage is installed
+  describe command('selinuxenabled') do
+    its('exist?') { should be true }
+    its('exit_status') { should eq 0 }
+  end
+
+  describe command('semanage') do
+    its('exist?') { should be true }
+  end
+
+  semanage_results = command('semanage login -l -n')
 
-  describe package("policycoreutils-python") do
-    it { should be_installed }
+  describe semanage_results do
+    its('stdout.lines') { should_not be_empty }
   end
 
-  semanage_results = command("semanage login -l").stdout.split("\n")
-  # Remove Header Row
-  semanage_results.shift
+  semanage_results.stdout.lines.each do |result|
+    login, seuser = result.split(/\s+/)
+
+    # Skip Blank Lines
+    next unless login
 
-  semanage_results.each do |result|
-    result = result.gsub(/\s_/m, ' ').strip.split(" ")
     # Next if for some reason we still have header row
-    next if ( result[0] == 'Login')
+    next if ( login == 'Login')
+
     # Next if root
-    next if ( result[0] == 'root')
-    # Skip Blank Lines
-    next if ( /\S/ !~  result[0])
-
-    if admin_logins.include? "#{result[0]}"
-      describe.one do
-        describe command("semanage login -l | grep #{result[0]}") do
-          its('stdout') { should match %r{sysadm_u} }
-        end
-        describe command("semanage login -l | grep #{result[0]}") do
-          its('stdout') { should match %{system_u} }
-        end
-        describe command("semanage login -l | grep #{result[0]}") do
-          its('stdout') { should match %r{staff_u} }
-        end
-      end
-    elsif non_admin_logins.include? "#{result[0]}"
-      describe.one do
-	      describe command("semanage login -l | grep #{result[0]}") do
-          its('stdout') { should match %r{user_u} }
-        end
-        if( result[0] == '__default__')
-          # all real users should be mapped to a context (i.e. user_u)
-          # but the system isn't forced to map them by default to a context
-          # This will enable defualt to be unconfined like it is by default
-	        describe command("semanage login -l | grep #{result[0]}") do
-            its('stdout') { should match %r{unconfined_u} }
-          end
-        end
-      end
-    # Case when account isn't documented
-    else
-      describe command("semanage login -l | grep #{result[0]}") do
-        its('stdout.strip') { should be_empty }
+    next if ( login == 'root')
+
+    describe "SELinux login #{login}" do
```

Comment:
Doesn't allow `system_u` to pass since it seems it is mapped to `system_u` normally (always?)

Reply:
@samcornwell It's mapped to `system_u` *by default*. There's no telling what has been changed by the user on their system and the vendor documentation is pretty explicit that users should only ever be mapped to `staff_u`.

Reply:
@trevor-vaughan  So, what I'm talking to say is that `system_u` is a special login, it represents the system itself. I've been told that it should not be constrained. We should just ignore it, sort of like how we are just ignoring root.

Reply:
@samcornwell Ah, I see the disconnect! The only thing that is returned by `semanage login -l` should be actual user and groups that can login to the system. Therefore, `system_u` (and usually `sysadm_u`) should not appear as a mapping in this file.

Reply:
@trevor-vaughan Ok we'll go with that then